src/esp/gfx/ShaderManager.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "ShaderManager.h"
+
+#include "esp/core/esp.h"
+#include "esp/gfx/Drawable.h"
+#include "esp/scene/SceneNode.h"
+
+namespace esp {
+namespace gfx {
+
+void setLightSetupForSubTree(scene::SceneNode& root,
+                             const Magnum::ResourceKey& lightSetup) {
+  scene::preOrderFeatureTraversalWithCallback<Drawable>(
+      root, [&lightSetup](Drawable& drawable) {
+        drawable.setLightSetup(lightSetup);
+      });
+}
```

Comment:
Is this a "breadth-first" traversal? For this particular operation I think a simple recursion without any explicit `std::stack` could be enough (please correct me if I'm wrong, of course).

In any case, various tree traversals would be useful to have directly in Magnum itself, I'll think about integrating them.

Reply:
This is a pre-order, depth first traversal (ie. callback executed on parent, then move to children from left to right). Recursion would be fine, but I don't think we gain anything from it since the code is already quite readable, and technically the iterative approach is slightly faster, although I doubt there would be any real difference unless the tree was huge.